test/decl/protocol/req/optional.swift

```diff
@@ -219,10 +219,14 @@ optional class optErrorClass { // expected-error{{'optional' modifier cannot be
 }
   
 protocol optErrorProtocol {
-  optional func foo(_ x: Int) // expected-error{{'optional' can only be applied to members of an @objc protocol}}
+  optional func foo(_ x: Int) // expected-error{{'optional' requirements are an Objective-C compatibility feature; add '@objc'}}
+}
+
+@objc protocol optObjcAttributeProtocol {
+  optional func foo(_ x: Int) // expected-error{{'optional' requirements are an Objective-C compatibility feature; add '@objc'}}
```

Comment:
To test the Fix-It I mentioned about, add {{2-2=@objc }} following the expected-error's }}.